### PR TITLE
feat: add autoware_api_utils package

### PR DESCRIPTION
**Pull request by tkimura4**
# Related Issue
https://github.com/autowarefoundation/autoware.universe/issues/1

# Description
Add autoware_api_utils packages

# How to review
Check the following item
- Readme

### CODE REVIEW

File: common/autoware_api_utils/include/autoware_api_utils/rclcpp/client.hpp [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+  Client(typename rclcpp::Client<ServiceT>::SharedPtr client, rclcpp::Logger logger)
+  : client_(client), logger_(logger)
+  {
+  }
+
+  AutowareServiceResult call(
+    const typename ServiceT::Request::SharedPtr & request,
+    const std::chrono::nanoseconds & timeout = std::chrono::seconds(2))
+  {
+    // TODO(Takagi, Isamu): use rosidl_generator_traits::to_yaml (galactic)
```
Comment (1222-takeshi) [inline common/autoware_api_utils/include/autoware_api_utils/rclcpp/client.hpp:44]: Could you remove TODO comment ?

File: common/autoware_api_utils/include/autoware_api_utils/rclcpp/client.hpp [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+      return {response_error("Internal service is not available."), nullptr};
+    }
+
+    auto future = client_->async_send_request(request);
+    if (future.wait_for(timeout) != std::future_status::ready) {
+      RCLCPP_INFO(logger_, "client timeout");
+      return {response_error("Internal service has timed out."), nullptr};
+    }
+
+    // TODO(Takagi, Isamu): use rosidl_generator_traits::to_yaml (galactic)
```
Comment (1222-takeshi) [inline common/autoware_api_utils/include/autoware_api_utils/rclcpp/client.hpp:58]: Could you remove TODO comment ?

File: common/autoware_api_utils/include/autoware_api_utils/rclcpp/service.hpp [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+  explicit Service(typename rclcpp::Service<ServiceT>::SharedPtr service) : service_(service) {}
+
+  template <typename CallbackT>
+  static auto wrap(CallbackT && callback, const rclcpp::Logger & logger)
+  {
+    auto wrapped_callback = [logger, callback](
+                              typename ServiceT::Request::SharedPtr request,
+                              typename ServiceT::Response::SharedPtr response) {
+      // TODO(Takagi, Isamu): use rosidl_generator_traits::to_yaml (galactic)
```
Comment (1222-takeshi) [inline common/autoware_api_utils/include/autoware_api_utils/rclcpp/service.hpp:36]: Could you remove TODO comment ?

**Comment by 1222-takeshi** [comment]
@tkimura4 
Could you fix colcon test ?

File: common/autoware_api_utils/include/autoware_api_utils/rclcpp/client.hpp (hunk 2 above)
Comment (tkimura4) [reply common/autoware_api_utils/include/autoware_api_utils/rclcpp/client.hpp]: Fixed at fec1c52

File: common/autoware_api_utils/include/autoware_api_utils/rclcpp/client.hpp (hunk 1 above)
Comment (tkimura4) [reply common/autoware_api_utils/include/autoware_api_utils/rclcpp/client.hpp]: Fixed at fec1c52

**Review by 1222-takeshi — APPROVED** [review]
LGTM

**Comment by taikitanaka3** [comment]
@tkimura4 
933def4 9cf9df6 these changes should be reverted

**Pull request merged**